rpcs3/Emu/Cell/Modules/cellSsl.cpp

```diff
@@ -66,12 +78,15 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 		{
 			if (!flagBits[i-1])
 				continue;
-			final.append(getCert(certPath, i));
+			// If we're loading cert 6 (the baltimore cert), then we need set that we're loading the 'normal' set of certs.
+			if (flagBits[5])
+				final.append(getCert(certPath, i, true));
+			else
+				final.append(getCert(certPath, i));
 		}
 
-		memset(buffer.get_ptr(), 0, size);
-		strncpy(buffer.get_ptr(), final.c_str(), size);
-		buffer.get_ptr()[size] = '\0';
+		memset(buffer.get_ptr(), '\0', size - 1);
+		memcpy(buffer.get_ptr(), final.c_str(), size);
```

Comment:
I think this is wrong as it will leave the string not null terminated? Before changes it got explicitly null terminated (out of bounds mind). Assuming it is a c-style string its probably also wrong as final size might be less than size?

Reply:
I would agree, however, for some reason, if you add the null terminated character at the end, TLoU crashes.
I'll change the memcpy to `final.size()` though, thanks for pointing that out.

Reply:
@paulsapps if `final.c_str()` fits into buffer, it'll be null terminated (since `c_str()` is null-terminated). It's strange, however, that setting last char of the buffer to `0` leads to a crash.
Possible causes: 1) TLoU passes wrong (decremented) value (either because of emulation inaccuracies or because it is what it does originally) 2) buffer is not supposed to be null-terminated 3) something else.

Btw, you say that this PR fixes #3408 but has anyone done any tests with the actual hw/lle to confirm the behavior?

Reply:
I don't think its strange because the old code of `buffer.get_ptr()[size] = '\0';` is writing one out of bounds no?